src/main/java/controllers/Ui.java

```diff
@@ -0,0 +1,152 @@
+
+            // try catch statement wraps around getCommand which may throw an error if
+            // command input does not adhere to command format
+            try {
+                //@formatter:off
+                switch (command.getCommand()) {
+
+                case LIST:
+                    // list all the current todos
+                    todosController.listTodos();
+                    break;
+
+                case EVENT:
+                    // add a new event to the model
+                    try {
+                        todosController = todosController.addEvent(command.getCommandArgs());
+
+                    } catch (Exception e) {
+                        greeting.printErrorMessage(e);
+                    }
+                    break;
+
+                case TODO:
+                    // add a new todo to the model
+                    try {
+                        todosController = todosController.addTodo(command.getCommandArgs());
+
+                    } catch (DukeBlankTaskException e) {
+                        greeting.printErrorMessage(e);
+                    }
+                    break;
+
+                case DEADLINE:
+                    // amend a current todo's deadline
+                    try {
+                        todosController = todosController.addDeadline(command.getCommandArgs());
+
+                    } catch (Exception e) {
+                        greeting.printErrorMessage(e);
+                    }
+                    break;
+
+                case DONE:
+                    // mark a todo as done
+                    try {
+                        todosController = todosController.markAsDone(command.getCommandArgs());
+
+                    } catch (DukeTaskIndexOutOfRangeException e) {
+                        greeting.printErrorMessage(e);
+                    }
+                    break;
+
+                case DELETE:
+                    // delete a todo from the list
+                    try {
+                        todosController = todosController.deleteTodo(command.getCommandArgs());
+
+                    } catch (Exception e) {
+                        greeting.printErrorMessage(e);
+                    }
+                    break;
+                    
+                case FIND:
+                    // find a todo with the relevant keyword
+                    todosController.findByKeyword(command.getCommandArgs());
+                    break;
+
+                case BYE:
+                    // break out of main function
+                    greeting.bye();
+
+                    // save current todosController tasks to local db before exit
+                    storage.saveTasksToLocalDatabase(todosController.getTodosList());
+
+                    // close scanner preventing mem leak
+                    sc.close();
+
+                    // return to end function
+                    return;
```

Comment:
Perhaps explain a bit more about the specific use of `return` since every other cases are using `break`?

Reply:
Thanks Mark for your feedback and kind words. I will take your feedback and implement the relevant changes necessary 👍